### ISSUE TITLE
Unintialized IND0/IND1 in RRTM

### ISSUE THREAD

**Issue by JS-WRF-SBM** [wrf-model/WRF#1531]
**Describe the bug**
In the radiation driver, specifically in the `module_ra_rrtm.F`, when calling the routine `subroutine TAUGB3()`, both of the table index IND0/IND1 becomes uninitialized and produce out-of-bound SEGFAULT.

**Traceback**
forrtl: severe (408): fort: (3): Subscript #1 of the array ABSA3 has value -2147483637 which is less than the lower bound of 1

Image              PC                Routine            Line        Source             
wrf.exe            00000000039A3C46  Unknown               Unknown  Unknown
wrf.exe            0000000002697B8D  module_ra_rrtm_mp        4794  module_ra_rrtm.f90
wrf.exe            0000000002709EE8  module_ra_rrtm_mp        6688  module_ra_rrtm.f90
wrf.exe            000000000263BAEE  module_ra_rrtm_mp        2230  module_ra_rrtm.f90
wrf.exe            00000000026372AD  module_ra_rrtm_mp        1988  module_ra_rrtm.f90
wrf.exe            0000000001D424C7  module_radiation_        1581  module_radiation_driver.f90
wrf.exe            0000000001EC0F2C  module_first_rk_s         254  module_first_rk_step_part1.f90
wrf.exe            0000000001536EC2  solve_em_                 897  solve_em.f90
wrf.exe            0000000001379F0C  Unknown               Unknown  Unknown
wrf.exe            00000000005787B7  Unknown               Unknown  Unknown
wrf.exe            0000000000413D91  Unknown               Unknown  Unknown
wrf.exe            0000000000413D4F  Unknown               Unknown  Unknown
wrf.exe            0000000000413CE2  Unknown               Unknown  Unknown
libc-2.17.so       00002ADA8F3693D5  __libc_start_main     Unknown  Unknown
wrf.exe            0000000000413BE9  Unknown               Unknown  Unknown

**Approx source code** (line #4794)
```
!!DIR$ VECTOR           
         print*,"IND0",IND0,"IND1",IND1                                                          
         DO 2000 IG = 1, NG3                                                     
            TAUG(NGS2+IG,LAY) = SPECCOMB *             & 
                (FAC000 * ABSA3(IND0,IG) +                        &                 
                 FAC100 * ABSA3(IND0+1,IG) +                    &                 
                 FAC010 * ABSA3(IND0+10,IG) +                  &                 
                 FAC110 * ABSA3(IND0+11,IG) +                  &                 
                 FAC001 * ABSA3(IND1,IG) +                        &                 
                 FAC101 * ABSA3(IND1+1,IG) +                    &                 
                 FAC011 * ABSA3(IND1+10,IG) +                  &                 
                 FAC111 * ABSA3(IND1+11,IG)) +                 &                 
                 COLH2O(LAY) *                                             &                 
                 (SELFFAC(LAY) * (SELFREFC3(INDS,IG) +  &                 
                 SELFFRAC(LAY) *                                          &                 
                 (SELFREFC3(INDS+1,IG) - SELFREFC3(INDS,IG))) +    &                 
                 FORFAC(LAY) * FORREFC3(IG))                                     &                 
                 + N2OMULT * ABSN2OAC3(IG)                                         
                 PFRAC(NGS2+IG,LAY) = FRACREFAC3(IG,JS) + FS *   & 
                 (FRACREFAC3(IG,JS+1) - FRACREFAC3(IG,JS))
```

**To Reproduce**
This is a relatively heavy MC3E testcase simulation which can be reproduce with the attached `namelist.output`
The above bug occurs after 10h of simulation (!), at the same time of writing wrfout_dO2* (although, not sure this is related).
Attached is an example of the traceback thrown by the compiler, and one pair of the rsl.* files.
Compiler: ifort (IFORT) 19.0.3.199 20190206
Copyright (C) 1985-2019 Intel Corporation.

[rsl_files_0000.gz](https://github.com/wrf-model/WRF/files/6787998/rsl_files_0000.gz)
[namelist_input.gz](https://github.com/wrf-model/WRF/files/6787999/namelist_input.gz)

**Comment by JS-WRF-SBM** [comment]
@davegill @dudhia 
For the mean time, should I switch for the RRTMG scheme in the `namelist.input`?

**Comment by weiwangncar** [comment]
@JS-WRF-SBM This should be posted on the Forum: https://forum.mmm.ucar.edu/. Thanks.

**Comment by JS-WRF-SBM** [comment]
@weiwangncar 
I do not understand why I should post on the Forum of all the users a clear bug in the model?
Can you clarify?

**Comment by honnorat** [comment]
This looks like a nasty bug.

Glancing at the code, I can't see any good reason why "Subscript #1 of the array ABSA3" (aka `IND0`) could be uninitialized or take the value -2147483637. It could be seen as the more reasonable '11' integer with the sign bit set (because 2^31-2147483637 = 11). But why your CPU would flip the bit sign ?

Is this bug replicable ?

**Comment by dudhia** [comment]
Yes, RRTMG is for sure more supported. We should phase out the old RRTM
longwave.

**Comment by dudhia** [comment]
It can be posted as an issue.

**Comment by weiwangncar** [comment]
@JS-WRF-SBM My point is that this scheme has been in the model from the very beginning, and we haven't seen any problems for a long time. Sometimes model set up or other errors in the model may trigger unexpected response from radiation schemes. By posting the problem on the Forum, we would like to see if there is any configuration issues that may be the cause of the problem, and be addressed before it is identified as a real issue or bug with the model.

**Comment by dudhia** [comment]
However, we did see a response on the issue, so maybe this is a way to have
a discussion too.

**Comment by JS-WRF-SBM** [comment]
@honnorat @weiwangncar @dudhia 

Thanks for your comments.

1. Well, the fact that this kind of uninitialized indices (both of indices IND0/IND1) appear after 10h of simulation is likely caused by a combination of `namelist.input` options and/or compiler reasons.
2. Immediately after posting here I've used a restart file from 09h:30m and tested the RRTMG scheme for both SW/LW. This revealed another table index (but obviously different from the previous one) that was causing the same type of SEGFAULT as shown above. 
3. This gave me strong reasoning to believe the compiler is the main reason, so now I'm running the whole run again with an updated Intel 2019u5 compiler. Obviously, the fact the simulation is relatively heavy and the bug occurs after 10h of integration (cloud time), limits the amount of testing I can do.
